### DISCUSSION TITLE
Dealing with Access Control

### DISCUSSION THREAD

**Discussion by tomcrane** [digirati-co-uk/iiif-canvas-panel#54] [Q&A]
What, if anything, should Canvas Panel do about Auth? Specifically, IIIF Auth.

Image services (and later, audio, video, images, and other files) may be subject to access control, and may have implementations of IIIF Auth 1.0 or its successor specs.

If you are building a viewer using CP, should it help you manage the Auth flow? Or is that simply not its concern?

For example, OpenSeadragon doesn't know anything about Auth but that doesn't stop you using it in a viewer that does... you need to make sure you have dealt with auth before you ask OSD to render a tile source; CP could be the same.

On the other hand, CP does know about all the resources on the canvas, and could _help_ you implement auth in your surrounding app by describing what does require auth on the Canvas.

For example, it could fire events prior to rendering an authed resource, allowing you to cancel that rendering if you know the user doesn't have access yet.

I don't think it should handle auth completely by itself. There are state management and UI choices that need to be made above the level of CP.

**Comment by stephenwf** [comment]
This will fit nicely into #83 when we do come to tackle authentication. I think too that we should follow the same pattern that we settle on for choice. I'm currently writing up the new slice of Vault (meta) and it's potential use-cases. I think this would be another ideal case for centralising the state for the auth flow.

I think canvas panel or vault should be able to store the state for the flow, and also provide methods/functions for progressing that flow. It may not provide any user interfaces, but make it easy to show them. 

Having this at the Vault level would also allow you to sort out any authentication before rendering canvas panel. At least that could be a choice you make as a developer

**Reply by stephenwf** [reply]
Also related to #92
